### PR TITLE
Homework 13

### PR DESCRIPTION
**Pull request by DimitryusX**

### CODE REVIEW

File: Homework13/Utils.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+};
+
+enum class Complexity {
+    EASY = 'e',
+    HARD = 'h',
+};
+
+unsigned int getDay();
+std::string toLowerCase(const std::string& input);
+int randomNumberGenerator(const int min, const int max);
```
Comment (bohdanpc) [inline Homework13/Utils.h:19]: const in parameters is redundant

File: Homework13/Utils.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+void compare(std::string first, std::string second, char* hidden, Complexity mode);
```
Comment (bohdanpc) [inline Homework13/Utils.h:20]: const std::string&

File: Homework13/Utils.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#pragma once
```
Comment (bohdanpc) [inline Homework13/Utils.cpp:1]: no need in cpp

File: Homework13/Utils.cpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+std::string toLowerCase(const std::string& input)
+{
+    std::string result = input;
+
+    for (char& c : result) {
+        c = std::tolower(c);
+    }
+
+    return result;
+}
```
Comment (bohdanpc) [inline Homework13/Utils.cpp:15]: more efficient to make in-place of the same container

File: Homework13/UI.h [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+#include "iostream"
```
Comment (bohdanpc) [inline Homework13/UI.h:2]: not needed here, include in cpp

File: Homework13/Time.h [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#pragma once
+#include <ctime>
```
Comment (bohdanpc) [inline Homework13/Time.h:2]: same: not needed here

File: Homework13/File.cpp [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+void getListOfWordsFromFile(std::ifstream& file, std::string list[])
+{
+    std::string line;
+    size_t i = 0;
+
+    while (std::getline(file, line)) {
+        if (line == "") continue;
+        list[i] += line;
```
Comment (bohdanpc) [inline Homework13/File.cpp:26]: why not simple = ?

File: Homework13/Homework13.cpp [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+
+    // Reset file cursor
+    file.clear();
+    file.seekg(0);
+
+    // Store list of words to array
+    std::string listOfWords[MAX_WORD_ARRAY_SIZE];
+    getListOfWordsFromFile(file, listOfWords);
+
+    file.close();
```
Comment (bohdanpc) [inline Homework13/Homework13.cpp:45]: The main is too big. Why not split it into more functions? At least file related stuff can be easily wrapped into function as well.

File: Homework13/Homework13.cpp [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+                }
+
+                std::cout << std::endl << "Incorrect input value, try again: ";
+            } while (true);
+
+            std::cout << "Now your mode level: " << getModelLevelText(mode) << std::endl;
+            std::cout << std::endl;
+        }
+
+        if (navigation == Menu::HELP)
```
Comment (bohdanpc) [inline Homework13/Homework13.cpp:172]: same here: at least every navigation case can be its own sub-function of the bigger algorithm

File: Homework13/File.cpp (hunk 7 above)
Comment (DimitryusX) [reply Homework13/File.cpp:26]: Clear empty strings

**Pull request merged**